ledger/participant-integration-api/src/main/scala/platform/index/ContractStoreBasedMaximumLedgerTimeService.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.index
+
+import com.daml.ledger.participant.state.index.v2.{
+  ContractState,
+  ContractStore,
+  MaximumLedgerTime,
+  MaximumLedgerTimeService,
+}
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.value.Value
+import com.daml.lf.value.Value.ContractId
+import com.daml.logging.LoggingContext
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.chaining.scalaUtilChainingOps
+
+class ContractStoreBasedMaximumLedgerTimeService(contractStore: ContractStore)
+    extends MaximumLedgerTimeService {
+
+  override def lookupMaximumLedgerTimeAfterInterpretation(
+      ids: Set[Value.ContractId]
+  )(implicit loggingContext: LoggingContext): Future[MaximumLedgerTime] = {
+    def goAsync(
+        maximumLedgerTime: Option[Timestamp],
+        contractIds: List[ContractId],
+    ): Future[MaximumLedgerTime] =
+      (maximumLedgerTime, contractIds) match {
+        case (result, Nil) =>
+          Future.successful(MaximumLedgerTime.from(result))
+
+        case (resultSoFar, contractId :: restContractIds) =>
```

Comment:
`otherContractIds` ?

Reply:
I trust the Englishman on this one :) :+1: